packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,263 @@
+import type { Table } from '..';
+import type { TableColumn } from '../../table-column/base';
+import type { TableRecord } from '../types';
+
+/**
+ * This class manages the layout of columns within a Table.
+ * @interal
+ */
+export class TableLayoutManager<TData extends TableRecord> {
```

Comment:
I didn't look at this file in any amount of detail. I think someone besides you should understand it and take a closer look but it needn't be every reviewer. I think I heard you mention that person was likely to be @mollykreis but I'm happy to step in and get an overview if that isn't true.